google/resource_container_node_pool.go

```diff
@@ -85,6 +87,34 @@ func resourceContainerNodePoolCreate(d *schema.ResourceData, meta interface{}) e
 		InitialNodeCount: int64(nodeCount),
 	}
 
+	if v, ok := d.GetOk("node_config"); ok {
+		nodeConfigs := v.([]interface{})
+		if len(nodeConfigs) > 1 {
```

Comment:
Instead of this check, let's set `MaxItems: 1` in the schema itself (and then we can get rid of this check in `resource_container_cluster.go` too)